pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -72,10 +249,255 @@ describe('updateFromCommandLine', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
+          });
+        });
+
+        describe('when both profiles exit, system locked, user unlocked', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 1));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
+          });
+        });
+
+        describe('when both profiles exist and are locked', () => {
+          beforeEach(() => {
+            mock = jest.spyOn(fs, 'readFileSync')
+              .mockImplementation(createMocker(2, 2));
+          });
+          afterEach(() => {
+            mock.mockRestore();
+          });
+          test('all fields are locked', () => {
+            settings.load();
+            verifyAllFieldsAreLocked(settings.getLockedSettings());
+          });
+        });
+      });
+    });
+  });
+
+  describe('lockableFields', () => {
+    test('flattens an object with only allowed-image settings', () => {
+      const lockedSettings = {
+        containerEngine: {
+          imageAllowList: {
+            enabled:  true,
+            patterns: ["Shouldn't see this"],
+          },
+        },
+      };
+      const expectedLockedFields = {
+        containerEngine: {
+          imageAllowList: {
+            enabled:  true,
+            patterns: true,
+          },
+        },
+      };
+      const calculatedLockedFields = settings.determineLockedFields(lockedSettings);
+
+      expect(calculatedLockedFields).toEqual(expectedLockedFields);
+    });
+    test('flattens a complex object', () => {
+      const lockedSettings = {
+        virtualMachine: {
+          memoryInGB: 2,
+          numberCPUs: 2,
+        },
+        containerEngine: {
+          imageAllowList: {
+            enabled:  true,
+            patterns: ["Shouldn't see this"],
+          },
+        },
+        kubernetes: { version: '1.2.3' },
+      };
+      const expectedLockedFields = {
+        virtualMachine: {
+          memoryInGB: true,
+          numberCPUs: true,
+        },
+        containerEngine: {
+          imageAllowList: {
+            enabled:  true,
+            patterns: true,
+          },
+        },
+        kubernetes: { version: true },
+      };
+      const calculatedLockedFields = settings.determineLockedFields(lockedSettings);
+
+      expect(calculatedLockedFields).toEqual(expectedLockedFields);
+    });
+    test('flattens an empty object', () => {
+      const lockedSettings = { };
+      const expectedLockedFields = { };
+      const calculatedLockedFields = settings.determineLockedFields(lockedSettings);
+
+      expect(calculatedLockedFields).toEqual(expectedLockedFields);
+    });
   });
 
   describe('getUpdatableNode', () => {
```

Comment:
(This comment goes with line 505, but GitHub won't let me enter it there…)
Hmm, should that be `blah.blah.blah`? We moved to dot-separated, right?

Edit: Huh, when I diffed with `-w` (i.e. told GitHub to skip whitspace) that showed up as part of the changes…

Reply:
This will make more sense in the next commit, when I move command-line processing to a separate module.